src/Endatix.Api/Endpoints/Auth/Register.cs

```diff
@@ -0,0 +1,45 @@
+
+using Endatix.Core.UseCases.Register;
+using FastEndpoints;
+using MediatR;
+
+namespace Endatix.Api.Endpoints.Auth;
+
+/// <summary>
+/// Endpoint for registering new user
+/// </summary>
+public class Register(IMediator mediator) : Endpoint<RegisterRequest, RegisterResponse>
+{
+    public override void Configure()
+    {
+        Post("/auth/register");
+        AllowAnonymous();
+        Summary(s =>
+        {
+            s.Summary = "Register a new user";
+            s.Description = "Creates a new user account in the Endatix application using the provided email and password.";
+            s.Responses[200] = "User has been successfully registered.";
+            s.Responses[400] = "Registration failed. Please check your input and try again.";
+            s.ExampleRequest = new RegisterRequest("user@example.com", "Password123!", "Password123!");
+        });
+    }
+
+    public override async Task HandleAsync(RegisterRequest request, CancellationToken cancellationToken)
+    {
+        var registerUserCommand = new RegisterCommand(request.Email, request.Password);
+        var result = await mediator.Send(registerUserCommand, cancellationToken);
+
+        if (!result.IsSuccess)
```

Comment:
Actually, I wanted to add these methods, but I have abandoned this idea as it was crazy as I was sprinting to get things done. If you are OK, let's leave it like this and we can try to change it over pair programming to review the pros/const.

My thinking when I decided to not use the extension method:
_I wanted to keep the logic in the endpoint and be able to express the logic and test it in the same method. I have tried with the endpoint, but I felt it will slow me down as I it occured to me I needed to refactor the Login and Register commands._

Reply:
I agree that with `ToEndpointResponse` it's tricky to test and that's the reason actually the endpoint handlers don't have test coverage.
I think the right approach is to first simplify the call to `ToEndpointResponse` (he have a task for it, I don't have it now but it's easy to find) and use it everywhere. Then easily cover all the endpoints with tests using the approach for tests generation we already have.
However, currently of much higher priority is how the product is used and what the overall experience is than refactoring this so I'm absolutely ok to leave it this way for now.